zone/corpse.cpp

```diff
@@ -2112,3 +2169,68 @@ std::vector<int> Corpse::GetLootList() {
 	}
 	return corpse_items;
 }
+
+void Corpse::SetRezTimer(bool initial_timer)
+{
+	if (!rezzable) {
+		if (corpse_rez_timer.Enabled()) {
+			corpse_rez_timer.Disable();
+		}
+		return;
+	}
+
+	IsOwnerOnline();
+
+	if (!is_owner_online && !initial_timer) {
+		if (corpse_rez_timer.Enabled()) {
+			corpse_rez_timer.Disable();
+		}
+		return;
+	}
+
+	if (corpse_rez_timer.Enabled() && !initial_timer) {
+		return;
+	}
+
+	if (initial_timer) {
+		uint32 timer = RuleI(Character, CorpseResTimeMS);
+		if (killed_by == Killed_DUEL) {
+			timer = RuleI(Character, DuelCorpseResTimeMS);
+		}
+		rez_time = timer;
+	}
+
+	if (rez_time < 1) {
+		// Corpse is no longer rezzable
+		CompleteResurrection(true);
+		return;
+	}
+	corpse_rez_timer.SetTimer(rez_time);
+}
+
+void Corpse::IsOwnerOnline()
+{
+	Client* client = entity_list.GetClientByCharID(GetCharID());
+
+	if (!client) {
+		uint32 account_id = database.GetAccountIDByChar(GetCharID());
+		client = entity_list.GetClientByAccID(account_id);
```

Comment:
What if the owner character is not online but they're online on another character? Do we care?

Reply:
We do care as the corpse rot timers will continue to tick down if the account is on.

We check if the current client is online, if not, we pull the account ID and check if any client is on using that account id.